### ISSUE TITLE
[Core] Perf regression

### ISSUE THREAD

**Issue by jjyao** [ray-project/ray#41695]
### What happened + What you expected to happen

```
REGRESSION 22.84%: single_client_tasks_sync (THROUGHPUT) regresses from 1161.670131632561 to 896.3011478159317 (22.84%) in 2.9.0/microbenchmark.json

REGRESSION 13.46%: multi_client_tasks_async (THROUGHPUT) regresses from 27211.51041454346 to 23547.65328133274 (13.46%) in 2.9.0/microbenchmark.json

REGRESSION 9.71%: client__tasks_and_put_batch (THROUGHPUT) regresses from 11415.752622212967 to 10306.863268325735 (9.71%) in 2.9.0/microbenchmark.json

REGRESSION 8.69%: multi_client_put_calls_Plasma_Store (THROUGHPUT) regresses from 12224.373147431208 to 11162.251480555622 (8.69%) in 2.9.0/microbenchmark.json
```

The regression all happened before the grpc upgrade.

### Versions / Dependencies

master

### Reproduction script

N/A

### Issue Severity

None

**Comment by rickyyx** [comment]
`single_client_tasks_sync`


1st regression: 10.25, from 1.2k -> 1k

fe6d5f44a5 [Train] Improve loggings for Train worker scheduling information (#40536)
8cfc894d9f [Core] Support Intel GPU (#38553)
89f18fbbe3 [core][serve] Disable flaky tests (#40643)
792e4e6cad [Data] Move `_resolve_paths_and_filesystem` to `util` file (#40304)
ec2eaf72a0 [Data][Docs] Cross-link XGBoost example in Ray Data examples (#40650)
f1a34b9f3e [cluster launcher] [vSphere Provider] Support GPU Ray nodes on vSphere (#40616)
fc98a5f286 [deflakey] Deflakey test_actor_advanced.py (#40636)
a10aaa9b3f [serve] Lazily construct `asyncio.Event` to avoid attaching to the wrong loop (#40629)
887eddd924 [RLlib] Updating codeblocks in RLlib part 1 (#37271)
b22ef7b159 Run test_basic and other minimal test separately (#40615)
be2606e1cd [serve] Remove `numpy` as a required dependency (#40565)
490f7bb869 split workflow into 2 shards (#40624)
99543b26cf [RLlib] De-flake rl_module/mobilenet test (#40621)
9c2990d0fd [release] Release logs for 2.7.1 (#40611)
469f4d296a [Doc] Update miscellaneous.rst to reflect deletion of feature flag 'use_ray_syncer' (#40208)
36b66e673d [ci] add missing gpu tag to serve doc gpu test (#40623)
94beb5eb3d [Core]Fix NullPointerException cause by raylet id is empty when get actor info in java worker (#40560)

**Comment by jjyao** [comment]
The regression is caused by https://github.com/ray-project/ray/pull/38553

**Comment by jjyao** [comment]
For `client__tasks_and_put_batch` the regression is caused by #40370

Before: https://buildkite.com/ray-project/release-tests-branch/builds/2389#_
After: https://buildkite.com/ray-project/release-tests-branch/builds/2390#_

**Comment by jjyao** [comment]
For `client__tasks_and_put_batch` and `multi_client_put_calls_Plasma_Store`, the regression is fixed by #41842

**Comment by jjyao** [comment]
For `multi_client_tasks_async`, it's also fixed by https://github.com/ray-project/ray/pull/41743. The remaining is noise.
